crates/devolutions-pedm-shell-ext/src/lib_win.rs

```diff
@@ -253,13 +275,16 @@ fn start_listener() {
                     match command {
                         ChannelCommand::Exit => break,
                         ChannelCommand::Elevate(path) => {
-                            let mut payload = resolve_lnk(&path).unwrap_or_else(|| LaunchPayload {
-                                executable_path: path.as_os_str().to_str().map(str::to_owned),
-                                command_line: None,
-                                working_directory: None,
-                                creation_flags: 0,
-                                startup_info: None,
-                            });
+                            let mut payload =
+                                resolve_lnk(&path)
+                                    .or_else(|| resolve_msi(&path))
```

Comment:
question: So this is where the magic happen? Do you think you could include an inline comment documenting what you wrote in your PR message?

>By inspecting elevated .msi files launched from Explorer, we see that Explorer invokes %systemroot%\system32\msiexec, with the command line "%systemroot%\system32\msiexec" /i "{path-to-msi}".
>
>We achieve the same in the PEDM module by using the same command if the file extension is .msi. The .msi extension is already being trapped by the shell extension, but previously we would call CreateProcess on the .msi causing "file is not a valid Win32 executable".

Thanks!

Reply:
Thanks to you!